Added classNames and removeAll options

tarekis opened the pull request:
I've added two additional optional options and the corresponding logic for this plugin:
# classNames

`classNames` is a passable optional option, it's type is a `array of strings`.

It sepcifies classes for each legend `li` element.
If a valid array is passed, which is as long as the number of series (or, if passed as long as `legendNames`), those classes will be appended dynamically to their corresponding elements.
This allows generating classes for specific elements and to style them via CSS.

A specific example is stacking charts, and hiding overlapping legends for positioning purposes.
# removeAll

`removeAll`  is a passable optional option, it's type is `boolean`.

It can be set to allow the user to remove every single entry of data in a chart.
This is not enabled by default to retain the developer's default, but enables a more flexible usage of the toggle funcionality depending on personal preference.

tarekis committed:
added classnames and removeall options

SpaceK33z commented:
Thanks for your PR! See my comments in the source code for some nitpicks.

Could you also write tests for the added functionality?

SpaceK33z commented on chartist-plugin-legend.js:
```diff
@@ -80,11 +82,19 @@
                 var li = document.createElement('li');
                 li.className = 'ct-series-' + i;
+                // Append specific class to a legend element, if viable classes are given
+                if (classNamesViable)
```
`if (classNamesViable) {`

SpaceK33z commented on chartist-plugin-legend.js:
```diff
@@ -106,19 +116,29 @@
-                            removedSeries= [];
-                            var seriesItems = Array.prototype.slice.call(legendElement.childNodes);
-                            seriesItems.forEach(function(item){
+                        if(!options.removeAll){
```
`if (`

SpaceK33z commented on chartist-plugin-legend.js:
```diff
@@ -106,19 +116,29 @@
+                             li.classList.add('inactive');
+                          }
+                             // Set all series as active.
+                          else
```
`else {`

SpaceK33z commented on chartist-plugin-legend.js:
```diff
@@ -106,19 +116,29 @@
+                             removedSeries = [];
+                             var seriesItems = Array.prototype.slice.call(legendElement.childNodes);
+                             seriesItems.forEach(function (item)
+                             {
```
`(item) {`

SpaceK33z commented on chartist-plugin-legend.js:
```diff
@@ -106,19 +116,29 @@
+                             });
+                          }
+                       }
+                       else{
```
`else {`

tarekis commented:
Sure, i'm adding the syntax sugar and tests soon!

tarekis committed:
syntax sugar

edited syntax for 39f30d52dd36c4a4181d2207a19d0fada3386274 based on https://github.com/CodeYellowBV/chartist-plugin-legend/pull/24

tarekis committed:
additional syntax sugar

addition to 4047549ec17b058bc720b6994f65967508ee52c5

tarekis committed:
test options classNames and removeAll

add tests for 39f30d52dd36c4a4181d2207a19d0fada3386274

tarekis committed:
describe classNames and removeAll

describe options passable as of 39f30d52dd36c4a4181d2207a19d0fada3386274

tarekis commented:
I've added tests now too, and updated the README. Please have a quick look if you have the time, thanks!

SpaceK33z commented:
Thanks! Looking good.

SpaceK33z commented:
Published as `v0.4.0`.

tarekis commented:
Happy to contribute!
